### ISSUE TITLE
VDiskController.create_new says the volume is created but sometimes its not.

### ISSUE THREAD

**Issue by kinvaris** [openvstorage/framework#1251]
Behavior: The healthcheck performs a `create_new` for a vdisk, waits 2 seconds then deletes it again. 
These calls are all performed synchronously to avoid races. 

But on a gig environment we observe the following output:
```
Volumedriver of vPool 'data03' seems to have runtime problems. Got `VDisk with name ovs-healthcheck-test-jTLnzvJIirvrLyHZ.raw not found!` while executing. 
``` 

![](http://pix.toile-libre.org/upload/original/1481181937.png)

This error is caused by the following call: `VDiskList.get_by_devicename_and_vpool`, in general its telling us that the vdisk was not found.

**Comment by khenderick** [comment]
Can you provide logs of some kind. At this point, it doesn't say that much about what's going on.

**Comment by khenderick** [comment]
I somehow suspect that you get a ```RuntimeError```:

```
In [20]: VDiskList.get_by_devicename_and_vpool('/foo/bar.raw', p)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-20-2eb4d0a4f0ac> in <module>()
----> 1 VDiskList.get_by_devicename_and_vpool('/foo/bar.raw', p)

/opt/OpenvStorage/ovs/dal/lists/vdisklist.pyc in get_by_devicename_and_vpool(devicename, vpool)
     64         if len(vds) > 0:
     65             if len(vds) != 1:
---> 66                 raise RuntimeError('Invalid amount of vDisks found: {0}'.format(len(vds)))
     67             return vds[0]
     68         return None

RuntimeError: Invalid amount of vDisks found: 2
```

**Comment by kinvaris** [comment]
@khenderick: I've checked the environment and it appears that this is not the case.
This is a code snippit where it goes wrong:
```
    @staticmethod
    def get_vdisk_by_name(vdisk_name, vpool_name):
        """
        Fetch disk partitions by disk guid
        :param vdisk_name: location of a vdisk on a vpool
                           (e.g. /mnt/vpool/test.raw = test.raw, /mnt/vpool/volumes/test.raw = volumes/test.raw )
        :type vdisk_name: str
        :param vpool_name: name of a existing vpool
        :type vpool_name: str
        :return: a vdisk object
        :rtype: ovs.dal.hybrids.vdisk
        """

        vpool = VPoolList.get_vpool_by_name(vpool_name)
        if vpool:
            vdisk = VDiskList.get_by_devicename_and_vpool('/{0}'.format(vdisk_name), vpool)
            if vdisk:
                return vdisk
            else:
                raise VDiskNotFoundError("VDisk with name `{0}` not found!".format(vdisk_name))
        else:
            raise VPoolNotFoundError("vPool with name `{0}` cannot be found!".format(vpool_name))
```

The vdisk on this line `vdisk = VDiskList.get_by_devicename_and_vpool('/{0}'.format(vdisk_name), vpool)` returns `None` and we throw a `VDiskNotFoundError`.

When observing this error the `VDiskNotFoundError` is caught by this line: https://github.com/openvstorage/openvstorage-health-check/blob/cc8ec332b065922dfbe8c1a40ce8336926e286e0/ovs/extensions/healthcheck/volumedriver/volumedriver_health_check.py#L167

**Comment by khenderick** [comment]
When a ```create_vdisk``` is executed, you'll receive the new vDisk's guid back. Can you for testing purposes load the vDisk object with that guid and print it out? I'd like to know it's properties and relation to the vPool. After that, can you give me the exact paramters you pass into the ```get_by_devicename_and_vpool``` call.

You pass in some "vdisk name", while the ```get_by_devicename_and_vpool``` doesn't require a name, but a devicename is needed. And you cannot specify the devicename upon creation, but merely a "hint" on what it should be (it can change it based on a set of requirements). I suspect the requested name cannot be used as a devicename and is thus causing the above.

**Comment by wimpers** [comment]
@kinvaris please test Kenneth's suggestions.

**Comment by khenderick** [comment]
High likely that this was caused by the issue resolved in #1390.

**Comment by JeffreyDevloo** [comment]
Closing it down as the linked ticket has also been solved.
Suspected that the healthcheck (burn it at the stake!) was also behind the issue which was the cause of #1390
